### PR TITLE
DC-178: document and test enrollment-checker analytics events

### PR DESCRIPTION
**Pull request by adbergen**
#### 🔗 Jira ticket
https://codeforamerica.atlassian.net/browse/DC-178

#### ✍️ Description

Instruments the CO enrollment checker with the three analytics events the SEBT Data Layer Dictionary calls for:

- `enrollment_check_start` — fired on the child-form page; carries `name` + `application` from page context.
- `enrollment_check_result` — fired when the results page loads; carries `enrollment_match_type` (page) + `sebt_eligible` (user).
- `enrollment_check_error` — fired when the submission API call fails; carries `error_code` (page).

**Stack note.** This PR is stacked on top of [DC-304](https://github.com/codeforamerica/sebt-self-service-portal/pull/312), which fixes `_trackEvent` to merge analytics-scoped `page.*` and `user.*` context into every event payload. Once DC-304 merges to main, this PR's base auto-updates and the diff becomes purely DC-178-specific.

**Why this PR is small.** The three trackEvent call sites were already wired into the EC code (`ChildFormPage.tsx`, `app/results/page.tsx`, `app/review/page.tsx`) — they just shipped empty `eventData` because of the bug DC-304 fixes. With DC-304's merge available, each event automatically carries the page/user fields the dictionary calls for, so this PR is verification + documentation:

- JSDoc on each event constant in `events.ts` documenting the expected payload contract.
- Two new tests in `ChildFormPage.test.tsx` asserting the data-layer merge contract for `enrollment_check_start`:
  1. The event carries `name` + `application` in the payload.
  2. The payload contains no child PII (`firstName`, `lastName`, `dateOfBirth`).

#### Privacy review

Per the ticket AC: "A brief privacy review confirms no child PII is included in analytics payloads."

Each event was traced through the data layer to identify what `_trackEvent` would merge in:

- **`enrollment_check_start`** — page context only (set by `DataLayerProvider`: `application`, `entry_source`, `environment`, `language`, `locale`, `name`, `flow`, `step`). No child fields touch `page.*` or `user.*` at fire time. Test pins this: child PII set on `EnrollmentProvider` state does not leak into the event payload.
- **`enrollment_check_result`** — adds `enrollment_match_type` (one of `exact` / `fuzzy` / `none`, derived from `status` field only) + `sebt_eligible` (boolean derived from `status` only). The child name/DOB fields in the API response are NOT read by the analytics derivation in `results/page.tsx`.
- **`enrollment_check_error`** — adds `error_code` (`RATE_LIMIT` or `SUBMISSION_ERROR`, derived from the error message text). Caller-provided error text never includes child PII; the derived code is one of two string literals.

End-to-end tests for the result + error events were drafted but the page-route imports hung vitest at module load — likely a transitive issue unrelated to this change. The merge contract itself is fully tested by DC-304's `data-layer.test.ts` ("merges analytics-scoped page and user fields into eventData"). The remaining gap is covered by manual QA — see the QA summary.

#### Manual QA checklist

See PR comment.

#### Mapped to taxonomy ACs

- ✅ Documented event taxonomy: SEBT Data Layer Dictionary (Google Sheet); event constants in `events.ts` cross-reference it.
- ✅ Frontend emits events: `ChildFormPage.tsx`, `app/results/page.tsx`, `app/review/page.tsx`.
- ✅ Match types: `enrollment_match_type` value enum covers `exact` / `fuzzy` / `none`.
- ✅ Eligible vs ineligible: `sebt_eligible` boolean.
- ✅ CTA tracking: existing `CtaTracker` in `DataLayerProvider` fires `cta_click` for Apply / Portal / income-calculator buttons via the `data-analytics-cta` attribute pattern.
- ✅ Privacy: no child PII — verified by test + per-event audit.

#### 🔗 Links to related PRs
- DC-304 (base of this PR): https://github.com/codeforamerica/sebt-self-service-portal/pull/312

#### ✅ Completion tasks

- [x] Added relevant tests
- [x] Meets acceptance criteria in ticket
- [ ] Configuration changes:
  - [ ] If new environment variables are added, update in Tofu
  - [ ] If new environment secrets are added, update in Tofu and set in AWS Secret Manager
  - [ ] If you're adding an appsetting, add it to the requisite example file, and update it in the AWS AppConfig
  - [ ] If appsetttings are changed, update in AWS AppConfig

[DC-304]: https://codeforamerica.atlassian.net/browse/DC-304?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by adbergen** [comment]
# QA Summary — DC-178

## What Changed

The CO Enrollment Checker now emits three analytics events with the fields the SEBT Data Layer Dictionary requires:

- **`enrollment_check_start`** — fires when the user reaches the child-form page. Carries page context (`name`, `application`, `flow`, `step`, etc.).
- **`enrollment_check_result`** — fires when the user lands on the results page. Carries the match type (`exact` / `fuzzy` / `none`) and an `sebt_eligible` boolean.
- **`enrollment_check_error`** — fires when the eligibility check API call fails. Carries an `error_code` (`RATE_LIMIT` or `SUBMISSION_ERROR`).

**No user-visible UI changes.** This is observability only — the checker looks and behaves identically. Verification happens via browser DevTools and the deployed analytics dashboards.

**No child PII appears in any event payload.** Children's first name, last name, date of birth, and school information do NOT cross the analytics boundary — verified by both manual audit and an automated test.

## Affected Areas

- CO Enrollment Checker — Child form page (the page where the user enters a child's information)
- CO Enrollment Checker — Review page (the page that lists children before submitting)
- CO Enrollment Checker — Results page (the page that shows eligibility outcomes)
- Analytics tools — Mixpanel / Amplitude / Siteimprove for the CO environment (whichever are configured for that deployment)

## Test Cases

### Happy Path

- [ ] **`enrollment_check_start` fires when the user opens the child-form page.**
  1. From the CO Enrollment Checker landing page, click through the disclaimer and reach the child form page.
  2. With DevTools open, run `window.digitalData.event` in the console and look for an entry where `eventName === 'enrollment_check_start'`.
  3. **Expected:** The event exists. Its `eventData` includes at least `name` (a page title like "Check") and `application` (the value `"sebt-enrollment-checker"`).

- [ ] **`enrollment_check_result` fires with `enrollment_match_type=exact` for a child who matches.**
  1. Walk the full checker flow and submit a child whose first name, last name, and date of birth match a record in the lower environment's test data.
  2. After the results page loads, run `window.digitalData.event` and find the `enrollment_check_result` entry.
  3. **Expected:** `eventData.enrollment_match_type === "exact"` and `eventData.sebt_eligible === true`.

- [ ] **`enrollment_check_result` fires with `enrollment_match_type=fuzzy` for a near-match.**
  1. Submit a child whose details closely (but not perfectly) match a record — e.g. a one-letter spelling variation in the last name.
  2. Inspect the latest `enrollment_check_result` event.
  3. **Expected:** `enrollment_match_type === "fuzzy"` and `sebt_eligible === true`.

- [ ] **`enrollment_check_result` fires with `enrollment_match_type=none` and `sebt_eligible=false` for a no-match.**
  1. Submit a child whose details do not match any record.
  2. Inspect the latest `enrollment_check_result` event.
  3. **Expected:** `enrollment_match_type === "none"` and `sebt_eligible === false`.

- [ ] **`enrollment_check_error` fires when the API call fails.**
  1. Block outbound traffic from the browser to the checker API endpoint (e.g. via DevTools "Block request URL") or trigger a deliberate backend error in the lower environment.
  2. Add a child, reach the review page, and click Submit.
  3. **Expected:** The user sees an inline error alert. `window.digitalData.event` contains an `enrollment_check_error` entry with `eventData.error_code === "SUBMISSION_ERROR"`.

- [ ] **`enrollment_check_error` fires with `error_code=RATE_LIMIT` on rate-limited submissions.**
  1. Submit multiple checks in rapid succession until the API returns a rate-limit error (the inline alert text references rate limiting).
  2. Inspect the latest `enrollment_check_error` event.
  3. **Expected:** `eventData.error_code === "RATE_LIMIT"`.

### Edge Cases

- [ ] **No child PII appears in any analytics event.**
  1. Submit a child with a recognizable test name (e.g. "PiiTestChild ZebraXylo") through the full flow.
  2. After completing the flow (or hitting an error), search the full `window.digitalData.event` array as JSON for the test name.
  3. **Expected:** Neither the first name, last name, date of birth, nor school name appears in any of the three event payloads.

- [ ] **Multiple children produce one result event with a single match type.**
  1. Add two children — one expected to match and one expected to fail.
  2. Reach the results page.
  3. **Expected:** Exactly one `enrollment_check_result` event fires. Its `enrollment_match_type` is `exact` if any child matches, `fuzzy` if any near-match exists, otherwise `none`. `sebt_eligible` is `true` when at least one match (exact or fuzzy) was found, `false` otherwise.

### Regression Checks

- [ ] **Existing checker flow works normally.**
  1. Submit a child end-to-end and confirm the results page shows the expected per-child status, the apply CTA, and the portal CTA.
  2. **Expected:** No new errors, no missing UI, no broken navigation.

- [ ] **CTA tracking continues to work.**
  1. On the results page, click any tracked CTA (e.g. "Apply now", "Sign in to portal", income-calculator link).
  2. Inspect `window.digitalData.event` for the matching `cta_click` event.
  3. **Expected:** The `cta_click` event fires with the CTA's `cta_id` / `cta_target` values, as before this change.

- [ ] **The DC portal (separate app) is unaffected.**
  1. Sign in to the DC portal and confirm dashboard, ID proofing, and CTA tracking all behave normally.
  2. **Expected:** No regressions — DC-178 is scoped to the CO Enrollment Checker.

## Test Credentials

Not applicable — the CO Enrollment Checker is a public, unauthenticated tool. No sign-in is required.

For the matching test cases, QA will need seeded enrollment records in the lower environment. Coordinate with the data team if you need fresh fixtures.

## Environment Notes

- Test on the CO Enrollment Checker app (not the SEBT Portal). Both apps deploy independently.
- DevTools is the primary verification tool: `window.digitalData.event` returns the in-memory event array; each entry has `eventName`, `eventData`, `timeStamp`, `scope`.
- For end-to-end vendor verification, the Mixpanel / Amplitude / Siteimprove dashboards' Live View is the fastest way to confirm payloads land downstream.
- No new env vars, secrets, or app-settings — nothing to update in Tofu, AWS Secret Manager, or AppConfig.
- This PR is stacked on DC-304's fix to the data layer's event-merge logic. If DC-304 has not yet merged to main, expect the events to fire with empty payloads in any environment that does not yet include DC-304.

File: src/SEBT.EnrollmentChecker.Web/src/features/enrollment/components/ChildFormPage.test.tsx [hunk 1]
```diff
@@ -61,4 +62,44 @@ describe('ChildFormPage', () => {
...
+
+    it('does not include child PII in the enrollment_check_start payload', () => {
+      new DataLayer('digitalData')
+
+      // Render in edit mode so a child with full PII is in EnrollmentProvider state.
+      render(<ChildFormPageInEditMode />, { wrapper })
+
+      const event = window.digitalData!.event.find(e => e.eventName === 'enrollment_check_start')
+      expect(event).toBeDefined()
+      const serialized = JSON.stringify(event!.eventData)
```
Comment (necampanini) [inline src/SEBT.EnrollmentChecker.Web/src/features/enrollment/components/ChildFormPage.test.tsx:120]: I could be wrong but I think this test passes with an empty `{}`

to be certain, we could assert the payload is non-empty and have its keys match an expected allow-list, or seed the PII into page scope and assert it gets filtered out

**Review by necampanini — CHANGES_REQUESTED** [review]
looks good, just the one check on a test requested

**Review by necampanini — APPROVED** [review]
